cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -341,13 +338,40 @@ func (p *dockerProject) Build(
 	}, nil
 }
 
+func useDotnetPublishForDockerBuild(serviceConfig *ServiceConfig) bool {
+	if serviceConfig.useDotNetPublishForDockerBuild != nil {
```

Comment:
It felt like we were calling this a lot and caching the value (to save the stat calls) was worthwhile - but I could be convinced otherwise.  I didn't have anything scientific here - I just had a breakpoint in this method while I was developing things and felt like it was called many times during the `up` flow.